src/goto-instrument/cover_instrument_branch.cpp

```diff
@@ -36,7 +39,7 @@ void cover_branch_instrumentert::instrument(
 
   if(
     i_it->is_goto() && !i_it->guard.is_true() &&
-    !i_it->source_location.is_built_in())
+    goal_filters(i_it->source_location))
```

Comment:
:bulb: I'd say, if the goal filter is independent from the two `if` blocks, just:
```
if(!goal_filters(i_it->source_location)) return;
```
before the two `if`s. It makes it easier to see that instrumentation halts in that case.

Reply:
The goal filter is the most expensive check - that's why it comes last. But I can certainly improve the code structure.